### PR TITLE
Run script allows empty docker_args

### PR DESCRIPTION
**Pull request by illicitonion**
Currently when using `bazel run` on a docker image, I get:
```
/[redacted].executable: line 281: docker_args[@]: unbound variable
```

Alternatively, we could just `set +u` before execing docker.

### CODE REVIEW

File: container/incremental_load.sh.tpl [hunk 1]
```diff
@@ -277,6 +277,18 @@ if [[ "%{run}" == "True" ]]; then
   # We can clean up before we exec docker, since the exit handler will no longer run.
   cleanup
 
+  # Bash treats empty arrays as unset variables for the purposes of `set -u` - work around this.
```
Comment (mindaugasrukas) [inline container/incremental_load.sh.tpl:280]: nit: to be clear, `set -u` is not an issue but a kinda runtime linter. So this is not a workaround but a bug fix. Could you rephrase the comment to look not like a workaround?

File: container/incremental_load.sh.tpl [hunk 2]
```diff
@@ -277,6 +277,18 @@ if [[ "%{run}" == "True" ]]; then
...
+  args=(%{run_statement})
+  if [[ ${#docker_args[@]} -gt 0 ]]; then
+    args+=("${docker_args[@]}")
+  fi
+  args+=("%{run_tag}")
+  if [[ ${#container_args[@]} -gt 0 ]]; then
+    args+=("${container_args[@]}")
+  else
+    args+=("-")
+  fi
```
Comment (mindaugasrukas) [inline container/incremental_load.sh.tpl:290]: Please use the original form of `"${container_args[@]:-}"` or remove the `else` part. As if it's not set, it is empty but not `-`:
For more details, please see https://tldp.org/LDP/abs/html/parameter-substitution.html

**Review by mindaugasrukas — CHANGES_REQUESTED** [review]

**Comment by illicitonion** [comment]
Thanks for the review @mindaugasrukas - updated, PTAL!

**Review by mindaugasrukas — APPROVED** [review]
LGTM

**Review by linzhp — APPROVED** [review]
